### PR TITLE
Feat/onboarding v2 UI components and interaction behaviours

### PR DESCRIPTION
**Pull request by sharpshooter90**
### Summary

composed the page for onboarding questionnaire 
- added pure component
- added smooth scroll to on option click
- SCSS styles
- search with mock data
- filterByCategory with mock data

#### Related Issues / PR's

[](https://github.com/SigNoz/engineering-pod/issues/1744)

#### Screenshots

image

### CODE REVIEW

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss [hunk 1]
```diff
@@ -0,0 +1,233 @@
+.setup-flow {
+	margin-top: 74px;
+	z-index: 1;
+
+	&__header {
+		background: rgba(11, 12, 14, 0.7);
```
Comment (YounixM) [inline frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:6]: use variable from the design tokens

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+		color: var(--Vanilla-400, #c0c1c3);
+		margin-bottom: 48px !important;
+		line-height: 1.5;
+	}
+
+	&__search {
+		margin-bottom: 24px !important;
+	}
+
+	&__radio-buttons {
```
Comment (YounixM) [inline frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:75]: remove commented code.

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+		cursor: pointer;
+		box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.19);
+
+		&:hover {
+			background: var(--Robin-500, #4e74f8);
+			box-shadow: 0 4px 8px rgba(58, 134, 255, 0.2);
+		}
+
+		&__text {
+			color: var(--Vanilla-400, #c0c1c3);
```
Comment (YounixM) [inline frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:111]: incorrect variable names

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+		}
+
+		&__radio-custom {
+			padding: 8px 16px;
+			font-size: 14px;
+		}
+	}
+}
+
+.background-effect {
```
Comment (YounixM) [inline frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:183]: Is this responsive ?

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss (hunk 2 above)
Comment (sharpshooter90) [reply frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss]: done

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss (hunk 4 above)
Comment (sharpshooter90) [reply frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:177]: There are few changes need to made post this feature is done with the functionality. I will fix this once the flow is functional.

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss (hunk 3 above)
Comment (sharpshooter90) [reply frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss]: There are few changes need to made post this feature is done with the functionality. I will fix this once the flow is functional.

File: frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss (hunk 1 above)
Comment (sharpshooter90) [reply frontend/src/pages/OnboardingPageV2/OnboardingPageV2.styles.scss:6]: There are few changes need to made post this feature is done with the functionality. I will fix this once the flow is functional.

**Pull request closed without merging**